Rambo, Ramps arduino/bugfix builds are broken

I have been ignoring them for a while, hoping that bugfix will get fixed at some point. But that doesn't seem to be happening.

I'm not sure the root cause, this is the error I see:
```
2021-04-28T12:26:58.5211094Z In file included from /github/workspace/Marlin/Marlin/src/lcd/dogm/marlinui_DOGM.h:31:0,
2021-04-28T12:26:58.5211955Z                  from /github/workspace/Marlin/Marlin/src/lcd/marlinui.cpp:126:
2021-04-28T12:26:58.5213719Z /github/workspace/Marlin/Marlin/src/lcd/dogm/HAL_LCD_class_defines.h: In member function 'void U8GLIB_64128N_2X_HAL::init(pin_t, pin_t, pin_t, pin_t, pin_t)':
2021-04-28T12:26:58.5215121Z /github/workspace/Marlin/Marlin/src/lcd/dogm/HAL_LCD_class_defines.h:37:13: error: 'init' is not a member of 'U8GLIB'
2021-04-28T12:26:58.5216465Z      U8GLIB::init(&u8g_dev_st7565_64128n_HAL_2x_sw_spi, (uint8_t)sck, (uint8_t)mosi, (uint8_t)cs, (uint8_t)a0, (uint8_t)reset);
```

That looks like it isn't getting U8glib, or it is getting the wrong version. I see this at the top though:

```
2021-04-28T12:23:42.3525944Z Install required libraries
2021-04-28T12:23:42.3526495Z Install libraries U8glib
```
...
```
2021-04-28T12:26:23.7210869Z Alternatives for U8glib.h: [U8glib U8glib@1.19.1]
2021-04-28T12:26:23.7211484Z ResolveLibrary(U8glib.h)
2021-04-28T12:26:23.7225877Z   -> candidates: [U8glib U8glib@1.19.1]
```

The ways forward:
- Ignore this, as it is in bugfix
- Open an issue in main Marlin. The responses I've heard before to issues like this start with "use platformio", but not from Scott. They might want to know.
- Disable the arduino builds for these boards, at least on bugfix. This would be an even stronger way to ignore the errors :)
- Look into it, and try to update whatever U8glib dependency we need. Maybe the issue is that they changed it, or froze it, and we are specifying the wrong one? We specify it here: https://github.com/V1EngineeringInc/MarlinBuilder/blob/main/.github/workflows/rambo.yml#L98

--- a/.github/workflows/rambo.yml
+++ b/.github/workflows/rambo.yml
@@ -95,7 +95,7 @@ jobs:
         platform-url: https://raw.githubusercontent.com/ultimachine/ArduinoAddons/master/package_ultimachine_index.json
         # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
         set-build-path: true
-        required-libraries: U8glib
+        required-libraries: U8glib-HAL
 
     - name: Zip up Marlin
       run: |

--- a/.github/workflows/rambo.yml
+++ b/.github/workflows/rambo.yml
@@ -84,25 +84,6 @@ jobs:
       run: |
         src/core/build-for-machine
 
-    - name: Checkout U8glib-HAL Marlin library
-      uses: actions/checkout@v2
-      with:
-        repository: MarlinFirmware/U8glib-HAL
-        ref: 0.4.4
-        path: CustomU8glibMarlin
-
-    - name: Build
-      uses: ArminJo/arduino-test-compile@v3
-      with:
-        sketch-names: Marlin.ino
-        # -fqbn in the first line of the arduino console log.
-        arduino-platform: arduino:avr,rambo:avr
-        arduino-board-fqbn: rambo:avr:rambo
-        # This is what you would put into the board manager url.
-        platform-url: https://raw.githubusercontent.com/ultimachine/ArduinoAddons/master/package_ultimachine_index.json
-        # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
-        set-build-path: true
-
     - name: Zip up Marlin
       run: |
         src/core/zip-marlin

--- a/.github/workflows/rambo.yml
+++ b/.github/workflows/rambo.yml
@@ -85,11 +85,11 @@ jobs:
         src/core/build-for-machine
 
     - name: Checkout U8glib-HAL Marlin library
-        uses: actions/checkout@v2
-        with:
-          repository: MarlinFirmware/U8glib-HAL
-          ref: master
-          path: CustomU8glibMarlin
+      uses: actions/checkout@v2
+      with:
+        repository: MarlinFirmware/U8glib-HAL
+        ref: master
+        path: CustomU8glibMarlin
 
     - name: Build
       uses: ArminJo/arduino-test-compile@v2

--- a/.github/workflows/rambo.yml
+++ b/.github/workflows/rambo.yml
@@ -92,7 +92,7 @@ jobs:
         path: CustomU8glibMarlin
 
     - name: Build
-      uses: ArminJo/arduino-test-compile@v2
+      uses: ArminJo/arduino-test-compile@v3
       with:
         sketch-names: Marlin.ino
         # -fqbn in the first line of the arduino console log.

--- a/.github/workflows/rambo.yml
+++ b/.github/workflows/rambo.yml
@@ -95,7 +95,7 @@ jobs:
         platform-url: https://raw.githubusercontent.com/ultimachine/ArduinoAddons/master/package_ultimachine_index.json
         # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
         set-build-path: true
-        required-libraries: U8glib-HAL
+        required-libraries: U8glib-HAL@~0.4.4
 
     - name: Zip up Marlin
       run: |

--- a/.github/workflows/rambo.yml
+++ b/.github/workflows/rambo.yml
@@ -84,6 +84,13 @@ jobs:
       run: |
         src/core/build-for-machine
 
+    - name: Checkout U8glib-HAL Marlin library
+        uses: actions/checkout@v2
+        with:
+          repository: MarlinFirmware/U8glib-HAL
+          ref: master
+          path: CustomU8glibMarlin
+
     - name: Build
       uses: ArminJo/arduino-test-compile@v2
       with:
@@ -95,7 +102,6 @@ jobs:
         platform-url: https://raw.githubusercontent.com/ultimachine/ArduinoAddons/master/package_ultimachine_index.json
         # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
         set-build-path: true
-        required-libraries: U8glib-HAL@~0.4.4
 
     - name: Zip up Marlin
       run: |

--- a/.github/workflows/rambo.yml
+++ b/.github/workflows/rambo.yml
@@ -88,7 +88,7 @@ jobs:
       uses: actions/checkout@v2
       with:
         repository: MarlinFirmware/U8glib-HAL
-        ref: master
+        ref: 0.4.4
         path: CustomU8glibMarlin
 
     - name: Build

--- a/.github/workflows/ramps.yml
+++ b/.github/workflows/ramps.yml
@@ -89,7 +89,7 @@ jobs:
         path: CustomU8glibMarlin
 
     - name: Build
-      uses: ArminJo/arduino-test-compile@v2
+      uses: ArminJo/arduino-test-compile@v3
       with:
         sketch-names: Marlin.ino
         # -fqbn in the first line of the arduino console log.

--- a/.github/workflows/ramps.yml
+++ b/.github/workflows/ramps.yml
@@ -89,7 +89,7 @@ jobs:
         arduino-board-fqbn: arduino:avr:mega:cpu=atmega2560
         # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
         set-build-path: true
-        required-libraries: U8glib
+        required-libraries: U8glib-HAL
 
     - name: Zip up Marlin
       run: |

--- a/.github/workflows/ramps.yml
+++ b/.github/workflows/ramps.yml
@@ -89,7 +89,7 @@ jobs:
         arduino-board-fqbn: arduino:avr:mega:cpu=atmega2560
         # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
         set-build-path: true
-        required-libraries: U8glib-HAL
+        required-libraries: U8glib-HAL@~0.4.4
 
     - name: Zip up Marlin
       run: |

--- a/.github/workflows/ramps.yml
+++ b/.github/workflows/ramps.yml
@@ -81,25 +81,6 @@ jobs:
       run: |
         src/core/build-for-machine
 
-    - name: Checkout U8glib-HAL Marlin library
-      uses: actions/checkout@v2
-      with:
-        repository: MarlinFirmware/U8glib-HAL
-        ref: 0.4.4
-        path: CustomU8glibMarlin
-
-    - name: Build
-      uses: ArminJo/arduino-test-compile@v3
-      with:
-        sketch-names: Marlin.ino
-        # -fqbn in the first line of the arduino console log.
-        arduino-platform: arduino:avr,rambo:avr
-        arduino-board-fqbn: rambo:avr:rambo
-        # This is what you would put into the board manager url.
-        platform-url: https://raw.githubusercontent.com/ultimachine/ArduinoAddons/master/package_ultimachine_index.json
-        # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
-        set-build-path: true
-
     - name: Zip up Marlin
       run: |
         src/core/zip-marlin

--- a/.github/workflows/ramps.yml
+++ b/.github/workflows/ramps.yml
@@ -85,7 +85,7 @@ jobs:
       uses: actions/checkout@v2
       with:
         repository: MarlinFirmware/U8glib-HAL
-        ref: master
+        ref: 0.4.4
         path: CustomU8glibMarlin
 
     - name: Build

--- a/.github/workflows/ramps.yml
+++ b/.github/workflows/ramps.yml
@@ -81,15 +81,24 @@ jobs:
       run: |
         src/core/build-for-machine
 
+    - name: Checkout U8glib-HAL Marlin library
+        uses: actions/checkout@v2
+        with:
+          repository: MarlinFirmware/U8glib-HAL
+          ref: master
+          path: CustomU8glibMarlin
+
     - name: Build
       uses: ArminJo/arduino-test-compile@v2
       with:
         sketch-names: Marlin.ino
         # -fqbn in the first line of the arduino console log.
-        arduino-board-fqbn: arduino:avr:mega:cpu=atmega2560
+        arduino-platform: arduino:avr,rambo:avr
+        arduino-board-fqbn: rambo:avr:rambo
+        # This is what you would put into the board manager url.
+        platform-url: https://raw.githubusercontent.com/ultimachine/ArduinoAddons/master/package_ultimachine_index.json
         # Save the .hex/.bin/whatever file. It should end up in $HOME/Marlin/
         set-build-path: true
-        required-libraries: U8glib-HAL@~0.4.4
 
     - name: Zip up Marlin
       run: |

--- a/.github/workflows/ramps.yml
+++ b/.github/workflows/ramps.yml
@@ -82,11 +82,11 @@ jobs:
         src/core/build-for-machine
 
     - name: Checkout U8glib-HAL Marlin library
-        uses: actions/checkout@v2
-        with:
-          repository: MarlinFirmware/U8glib-HAL
-          ref: master
-          path: CustomU8glibMarlin
+      uses: actions/checkout@v2
+      with:
+        repository: MarlinFirmware/U8glib-HAL
+        ref: master
+        path: CustomU8glibMarlin
 
     - name: Build
       uses: ArminJo/arduino-test-compile@v2

